cumulus/ansible/tasks/providers/base.py

```diff
@@ -0,0 +1,48 @@
+class InstanceState():
+    PENDING = "pending"
+    RUNNING = "running"
+    SHUTTINGDOWN = "shutting-down"
+    TERMINATED = "terminated"
+    STOPPING = "stopping"
+    STOPPED = "stopped"
+
+
+class Provider(object):
+    __provider_registry__ = {}
```

Comment:
Should this be ClusterProvider? InstanceProvider? CloudProvider?  Provider seems overly generic to me.